docs/developer/developer.adoc

```diff
@@ -0,0 +1,357 @@
+:toc: preamble
+:numbered:
+
+This part of documentation is meant to serve mainly to developers who want to
+contribute to OpenSCAP, help to fix bugs, or take an advantage of
+the OpenSCAP library and create own projects on top of it.
+
+toc::[]
+
+== Building OpenSCAP on Linux
+If you want to build the ```libopenscap``` library and the ```oscap``` tool from
+the {oscap_git}[source code] then follow these instructions:
+
+. *Get the source code*
++
+Choose *1a* or *1b* depending on whether you want sources from a release tarball or the git repository.
+
+.. Use a release tarball:
+
+ # replace ${version} with the desired version
+ wget https://github.com/OpenSCAP/openscap/releases/download/${version}/openscap-${version}.tar.gz
+ tar -xzpf openscap-${version}.tar.gz
+ cd openscap-${version}
++
+**OR**
+
+.. Use fresh sources from git repository.
+
+ $ git clone https://github.com/OpenSCAP/openscap.git
+ $ cd openscap
++
+
+. *Get the build dependencies*
++
+--
+To build the library you will also need the following build dependencies.
+Build dependencies may vary depending on enabled features (by the ```cmake``` command).
+By default, OpenSCAP requires these dependencies to build:
+
+* CMake
+* DBus
+* SWIG
+* libxml2
+* librpm
+* libselinux
+* libgcrypt
+* pcre
+* python3
+* perl
+* libcurl
+* libxslt
+* libtool
+* perl-XML-XPath
+
+(some of these are optional, if they are not detected, openscap will be compiled
+without respective optional features).
+
+On RHEL / Fedora / CentOS, the command to install these packages is:
+
+----
+sudo yum install \
+cmake dbus-devel GConf2-devel libacl-devel libblkid-devel libcap-devel libcurl-devel \
+libgcrypt-devel libselinux-devel libxml2-devel libxslt-devel make openldap-devel \
+pcre-devel perl-XML-Parser perl-XML-XPath perl-devel python-devel rpm-devel swig \
+bzip2-devel gcc-c++
+----
+
+On Ubuntu 16.04, the command to install these packages is:
+
+----
+sudo apt-get install -y cmake libdbus-1-dev libdbus-glib-1-dev libcurl4-openssl-dev \
+libgcrypt20-dev libselinux1-dev libxslt1-dev libgconf2-dev libacl1-dev libblkid-dev \
+libcap-dev libxml2-dev libldap2-dev libpcre3-dev python-dev swig libxml-parser-perl \
+libxml-xpath-perl libperl5.22 libbz2-dev librpm-dev g++
+----
+
+When you have all the build dependencies installed you can build the library.
+--
+
+. *Build the library*
++
+Run the following commands to build the library:
+
+ $ cd build/
+ $ cmake ../
+ $ make
+
+
+. *Run the tests*
++
+After building the library you might want to run library self-checks. To do
+that you need to have these additional packages installed:
+
+ wget lua which procps-ng initscripts chkconfig sendmail
++
+and it is also required to have `sendmail` service running on the system:
+
+ $ systemctl start sendmail.service
++
+Now you can execute the following command to run library self-checks:
+
+ $ ctest
++
+Note: If you want to run `make distcheck` you will also need to install
```

Comment:
I don't think that `make distcheck` is relevant in the `master` branch.

Reply:
Good point. In fact, this paragraph speaks about installing `asciidoctor`. `asciidoctor` is a tools that converts `.adoc` files to HTML, including this document. Generating HTML docs used to be a part of `make dist` in `maint-1.2`. We forgot to port it over to `master` when we have ported OpenSCAP to CMake. See https://github.com/OpenSCAP/openscap/issues/1083
Actually, I think we need to mention `asciidoctor` at other, place, most likely in the "Install" section.